### ISSUE TITLE
Please enforce a LANG as needed for `printf "%.0f\n" "$RAMGB"` in `storage-benchmark`

### ISSUE THREAD

**Issue by pcfe** [RedHatSatellite/satellite-support#80]
this bit in `storage-benchmark`:

```bash
RAMGB=`grep MemTotal /proc/meminfo  | awk '{s+=$2} END {print s / 1000000}'`
# Convert to integer
RAMGB=`printf "%.0f\n" "$RAMGB"`
```
does not work with some LANG setings (e.g. en_DK.UTF-8)

example:
```bash
[root@satellite ~]# echo $LANG
en_DK.UTF-8
[root@satellite ~]# ./storage-benchmark /var/lib/pulp/fio-testing/
./storage-benchmark: line 79: printf: 26.48: invalid number
[root@satellite ~]# LANG=en_US.UTF-8 ./storage-benchmark /var/lib/pulp/fio-testing/
This test creates a test file that is double (2X) the size of this system's
RAM in GB. This benchmark will create a test file of size.. 

52 Gigabytes

.. in the: [/var/lib/pulp/fio-testing//storage-benchmark] location. This is to ensure that the test utilizes
a combination of cached and non-cached data during the test.

**** WARNING! Please verify you have enough free space to create a 52 GB
file in this location before proceeding. 

Do you wish to proceed? (Y/N) 
```

**Comment by andreamtp** [comment]
Same problem here: pushed PR #82 to fix it.

**Comment by pablomh** [comment]
Hi @pcfe and @andreamtp,

Would you be OK with the approach suggested in https://github.com/RedHatSatellite/satellite-support/pull/88/files#r1532703028? In my testing it covers the `en_DK.UTF-8` usecase, but I'd love confirmation.

**Comment by pablomh** [comment]
This is what I'd like to commit:

`RAMGB=$(awk '/^MemTotal:/ {printf "%d\n", $2 / 1024 / 1024}' /proc/meminfo)`

**Comment by pcfe** [comment]
> Would you be OK with the approach suggested in https://github.com/RedHatSatellite/satellite-support/pull/88/files#r1532703028? In my testing it covers the `en_DK.UTF-8` usecase, but I'd love confirmation.

Because the planet has so many languages and I only speak a few, I asked to set LANG explicitly in your script to meet the needs of the script.
While the suggestion from last week¹ sounds sane to me, that does not mean it will cover any LANG setting a user can have

If you are confident your awk construct works in any language, sure use what you suggested.
But TBH I fail to grok why you do not simply set the language in your script to ensure the calculation always works.



¹ https://github.com/RedHatSatellite/satellite-support/pull/88/files#r1537771032

**Comment by pablomh** [comment]
Hi!

I've done this testing to verify that the proposed solution was sound.

After installing all the locales available to my RHEL 9.7 system:
```
# dnf install -y langpacks-*
```

I run the following:
```
# locale -a | while read LOCALE; do LANG=$LOCALE printf "%.0f\n" "$RAMGB" >/dev/null 2>&1 || echo "Error with locale: $LOCALE"; done
Error with locale: ast_ES
Error with locale: ast_ES.utf8
Error with locale: be_BY
Error with locale: be_BY@latin
Error with locale: be_BY.utf8
Error with locale: bg_BG
Error with locale: bg_BG.utf8
Error with locale: br_FR
Error with locale: br_FR@euro
. . .
Error with locale: tr_CY.utf8
Error with locale: tr_TR
Error with locale: tr_TR.utf8
Error with locale: uk_UA
Error with locale: uk_UA.utf8
Error with locale: vi_VN
#
```

And the new code:
```
# locale -a | while read LOCALE; do LANG=$LOCALE awk '/^MemTotal:/ {printf "%d\n", $2 / 1000 / 1000}' /proc/meminfo >/dev/null 2>&1 || echo "Error with locale: $LOCALE";  done
#
```
